tools/toolchain_profiler.py

```diff
@@ -0,0 +1,220 @@
+import subprocess, os, time, sys, tempfile
+
+original_sys_exit = sys.exit
```

Comment:
all this code up to the main `if` could go in the `if`, right? we don't need it when not profiling.